client/src/components/categorical/value/index.js

```diff
@@ -22,11 +22,20 @@ import { labelPrompt, isLabelErroneous } from "../labelUtil";
 import MiniHistogram from "../../miniHistogram";
 import MiniStackedBar from "../../miniStackedBar";
 
+const VALUE_HEIGHT = 11;
+const LEFT_MARGIN = 60;
+const CHECKBOX = 26;
+const CELL_NUMBER = 50;
+const ANNO_MENU = 26;
+const LABEL_MARGIN = 16;
+const CHART_WIDTH = 100;
+const CHART_MARGIN = 24;
```

Comment:
What is the performance gain from bringing these out of the class?  I opted to leave them in-line to increase readability b/c I thought the visual pros outweighed the performance.

Reply:
Ah, I lift them out of the `render()` function because the new methods `this.renderMiniStackedBar()` and `this.renderMiniHistogram()` need access to a few of these constants, so they can't be in `render()` function anymore. But I should put the rest back in `render()` for readability!

Thanks for raising this!

Reply:
Just moved most of the constants back 😄 PTAL again! Thank you 🙏